addons/l10n_it_edi/tests/test_res_company.py

```diff
@@ -0,0 +1,44 @@
+from odoo import Command
+from odoo.tests import tagged
+from odoo.addons.l10n_it_edi.tests.common import TestItEdi
+from odoo.tests.common import Form
+
+@tagged('post_install_l10n', 'post_install', '-at_install')
+class TestResCompany(TestItEdi):
+
+    def test_tax_representative(self):
+        """ Test to check for l10n_it_edi_tax_representive if l10n_it_has_tax_representative is not set."""
+
+        self.company.write({
+            "l10n_it_tax_representative_partner_id": self.italian_partner_a.id,
+            "l10n_it_has_tax_representative": True,
+        })
+
+        with Form(self.company) as company:
+            company.l10n_it_has_tax_representative = False
+
+        invoice = self.env['account.move'].with_company(self.company).create({
+            'move_type': 'out_invoice',
+            'invoice_date': '2024-01-31',
+            'invoice_date_due': '2024-01-31',
+            'partner_id': self.italian_partner_no_address_codice.id,
+            'invoice_line_ids': [
+                Command.create({
+                    'name': 'invoice_line',
+                    'price_unit': 100.00,
+                    'tax_ids': [Command.set(self.default_tax.ids)],
+                }),
+            ],
+        })
+
+        invoice.action_post()
+        self._assert_export_invoice(invoice, 'tax_representative_unchecked.xml')
+
+    def test_tax_representative_inverse(self):
+        """Test to check l10n_it_has_tax_representative if l10n_it_tax_representative is set"""
+
+        self.company.write({
+            "l10n_it_tax_representative_partner_id": self.italian_partner_a,
+        })
+
+        self.assertTrue(self.company.l10n_it_has_tax_representative)
```

Comment:
```suggestion
    def test_tax_representative(self):
        """ """
        # Check l10n_it_has_tax_representative if l10n_it_tax_representative is set
        self.company.l10n_it_tax_representative_partner_id = self.italian_partner_a.id
        self.assertTrue(self.company.l10n_it_has_tax_representative)
        
        # Test to check for l10n_it_edi_tax_representive if l10n_it_has_tax_representative is not set.
        company.l10n_it_has_tax_representative = False
        invoice = self.env['account.move'].with_company(self.company).create({
            'move_type': 'out_invoice',
            'invoice_date': '2024-01-31',
            'invoice_date_due': '2024-01-31',
            'partner_id': self.italian_partner_no_address_codice.id,
            'invoice_line_ids': [
                Command.create({
                    'name': 'invoice_line',
                    'price_unit': 100.00,
                    'tax_ids': [Command.set(self.default_tax.ids)],
                }),
            ],
        })

        invoice.action_post()
        self._assert_export_invoice(invoice, 'tax_representative_unchecked.xml')
```
Less cost, still readable imo